node/alphanet/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+extern crate serde;
+extern crate serde_derive;
+
+use std::{cmp, env, thread};
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::str::FromStr;
+use std::time::{Duration, Instant};
+
+use env_logger::Builder;
+use futures::future::Future;
+use futures::sink::Sink;
+use futures::sync::mpsc;
+use tokio::timer::Delay;
+
+use chain::ChainPayload;
+use network::nightshade_protocol::{spawn_consensus_network, start_peer};
+use nightshade::nightshade_task::{Control, spawn_nightshade_task};
+use primitives::aggregate_signature::{BlsPublicKey, BlsSecretKey, get_bls_key_pair};
+use primitives::hash::hash_struct;
+use primitives::network::PeerInfo;
+use primitives::signature::{get_key_pair, PublicKey, SecretKey};
+use primitives::types::AccountId;
+
+#[allow(dead_code)]
+fn configure_logging(log_level: log::LevelFilter) {
+    let internal_targets = vec![
+        "nightshade",
+        "alphanet",
+    ];
+    let mut builder = Builder::from_default_env();
+    internal_targets.iter().for_each(|internal_targets| {
+        builder.filter(Some(internal_targets), log_level);
+    });
+
+    let other_log_level = cmp::min(log_level, log::LevelFilter::Info);
+    builder.filter(None, other_log_level);
+
+    if let Ok(lvl) = env::var("RUST_LOG") {
+        builder.parse(&lvl);
+    }
+    if let Err(e) = builder.try_init() {
+        warn!(target: "client", "Failed to reinitialize the log level {}", e);
+    }
+}
+
+// TODO: Explain this function and their arguments
+fn run_node(authority: usize,
+            num_authorities: usize,
+            mut authorities: Vec<AccountId>,
+            boot_nodes: Vec<PeerInfo>,
+            public_keys: Vec<PublicKey>,
+            secret_key: SecretKey,
+            bls_public_keys: Vec<BlsPublicKey>,
+            bls_secret_key: BlsSecretKey,
+) {
+    let node_task = futures::lazy(move || {
+        info!(target: "alphanet", "node {}: Main task", authority);
+
+        // 1. Initialize peer manager
+
+        let account_id = authorities[authority].clone();
+
+        let (pm, _out_msg_tx, inc_msg_rx) =
+            start_peer(authority, num_authorities, Some(account_id), boot_nodes);
+
+        // Wait for all channels become available
+        while pm.count_ready_channels() < num_authorities - 1 {
+            thread::sleep(Duration::from_millis(50));
+        }
+
+        info!(target: "alphanet", "node: {} ready channels: {:?}", authority, pm.count_ready_channels());
+
+        // 2. Initialize NightshadeTask. Create proposals.
+
+        // TODO: Each participant should propose different chain payloads
+        let payload: ChainPayload = ChainPayload {
+            transactions: vec![],
+            receipts: vec![],
+        };
+
+        let (inc_gossip_tx, inc_gossip_rx) = mpsc::channel(1024);
+        let (out_gossip_tx, out_gossip_rx) = mpsc::channel(1024);
+        let (consensus_tx, _consensus_rx) = mpsc::channel(1024);
+
+        let control_tx = spawn_nightshade_task(
```

Comment:
Let us adopt the following convention:
If we have two components A and B that are both owned by component C, such that A and B are connected using channel (Receiver, Sender), then we construct (Receiver, Sender) in C and pass the corresponding ends to A and B.